### ISSUE TITLE
Show marriages in person trees

### ISSUE THREAD

**Issue by jat255** [bartfeenstra/betty#938]
Currently, the ancestor/descendent tree only shows direct ancestor/descendant relationships. In the case of multiple marriages (or even just one), it would be nice to see spousal relationships, and the children that arose from those relationships.

**Comment by bartfeenstra** [comment]
I spent some think thinking about this and exploring algorithms. Long story short: there are very, very few existing tools that let us lay out a family tree.

A simple family tree is any number of parents and any number of children for any given person. Assuming no cycles/diamonds (people aren't both someone's parent and grandparent, for example), this makes what is called **directed** and **acyclic** graph, also known as a simple tree, but 'with multiple roots'. These are fairly easily laid out.

If we want to support cycles (Oedipus complexes, or any European monarchy), or add spouses, the graph becomes more complex. Cycles mean people can be associated with others in different parts of the graph (meaning we very likely have to draw crossed lines), and spouses introduce a new concept of people appearing next to the person's in question. What if there are more than 2 spouses? Where does everyone go in the graph? What if the spouses' parents must be shown? Then the spouses musy either appear far away from their partners, or from their parents.

In short, adding additional relationships to the graphs makes the work significantly more complex, a task for which not enough maintainers with enough knowledge and time are available for.

While this is a perfectly valid feature request, and I'd love for it to be implemented, I am closing this issue because there is no chance this **can** be done in the near future.
